lib/hsa/hsaBeamformHFBOutput.cpp

```diff
@@ -0,0 +1,61 @@
+#include "hsaBeamformHFBOutput.hpp"
+
+using kotekan::bufferContainer;
+using kotekan::Config;
+
+REGISTER_HSA_COMMAND(hsaBeamformHFBOutputData);
+
+hsaBeamformHFBOutputData::hsaBeamformHFBOutputData(Config& config, const string& unique_name,
+                                                   bufferContainer& host_buffers,
+                                                   hsaDeviceInterface& device) :
+    hsaCommand(config, unique_name, host_buffers, device, "hsaBeamformHFBOutputData", "") {
+    command_type = gpuCommandType::COPY_OUT;
+
+    network_buffer = host_buffers.get_buffer("network_buf");
+    register_consumer(network_buffer, unique_name.c_str());
+    output_buffer = host_buffers.get_buffer("beamform_hfb_output_buf");
+    register_producer(output_buffer, unique_name.c_str());
+
+    network_buffer_id = 0;
+    output_buffer_id = 0;
+    output_buffer_execute_id = 0;
+    output_buffer_precondition_id = 0;
+}
+
+hsaBeamformHFBOutputData::~hsaBeamformHFBOutputData() {}
+
+int hsaBeamformHFBOutputData::wait_on_precondition(int gpu_frame_id) {
+    (void)gpu_frame_id;
+    uint8_t* frame =
+        wait_for_empty_frame(output_buffer, unique_name.c_str(), output_buffer_precondition_id);
+    if (frame == NULL)
+        return -1;
+    output_buffer_precondition_id = (output_buffer_precondition_id + 1) % output_buffer->num_frames;
```

Comment:
You should also do a `wait_for_full_frame()` on the network buffer here.   It's not strictly needed, but it helps avoid potential bugs if this is used in different configs, and improves the internal statistics.